winit-core/src/event.rs

```diff
@@ -1010,6 +1030,242 @@ pub type AxisId = u32;
 /// Identifier for a specific button on some device.
 pub type ButtonId = u32;
 
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
+pub struct ToolState {
```

Comment:
non_exhaustive

Reply:
Can't make it non_exhaustive as I need to construct it in winit-web